crates/sui-framework/packages/move-stdlib/sources/macros.move

```diff
@@ -0,0 +1,71 @@
+
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module holds shared implementation of macros used in `std`
+module std::macros {
+    public(package) macro fun num_max($x: _, $y: _): _ {
+        let x = $x;
+        let y = $y;
```

Comment:
can I ask the reason for the values assigned to locals? is that needed for reasons that are currently escaping me?

Reply:
macro args are call-by-expression. So if I did not do this, the argument expression would potentially be evaluated more than once